### ISSUE TITLE
Terminal is starting up at a size that's different from initialRows/initialCols 

### ISSUE THREAD

**Issue by 0xabu** [microsoft/terminal#2061]
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18362.264]
Windows Terminal version (if applicable): 8ffff8ea 0.2.1991.0
Any other software? WSL
```

# Steps to reproduce

The relevant-looking parts of profiles.json are:

```
{
    "globals" : 
    {
        "defaultProfile" : "{c6eaf9f4-32a7-5fdc-b5cf-066e8a4b1e40}",
        "initialCols" : 80,
        "initialRows" : 40,
    },
    "profiles" : 
    [
        {
            "commandline" : "wsl.exe -d Ubuntu-18.04",
            "fontFace" : "DejaVu Sans Mono for Powerline",
            "fontSize" : 9,
            "guid" : "{c6eaf9f4-32a7-5fdc-b5cf-066e8a4b1e40}",
            "padding" : "0, 0, 0, 0"
        }
    ]
```

# Expected behavior

I should see an 80x40 terminal.

# Actual behavior

I get an 80x37 terminal, with extra blank horizontal space after the last usable column:

![image](https://user-images.githubusercontent.com/13037682/61648053-84b98000-ac63-11e9-896c-453cee71e02a.png)

**Comment by culcube** [comment]
There appears to be two separate issues here.

1. Created terminal windows are consistently 3 rows smaller than the value for initialRows.
e.g. initialRows 27 gives a terminal size of height 24
Confirmed using powershell's "$Host.UI.RawUI.WindowSize.Height" and Ubuntu bash's "stty size" in their respective tabs.

2. Extra space appears to the right and bottom of the terminal window, but this isn't part of the terminal's reported size, and doesn't affect output only how the windows looks (at least when checked using an 80x24 (i.e. setting an 80x27, see point 1) terminal window).
I suspect the window is being drawn at the maximum size it can be without becoming an extra column wider or row taller, Perhaps the pixel size is calculated and 1 is subtracted to avoid a fence-post error, but this subtraction is from the size in pixels, and not the the number of rows or columns.

I can't see a style guide for Bug reporting - is the right thing to do to create two new bugs referencing this one (and then close this one)? To spin one issue out into a new bug and keep this bug open for the other one?

**Comment by culcube** [comment]
a

**Comment by zadjii-msft** [comment]
@culcube I'm fairly certain this is all due to the same root cause, so they can all stay in this issue. I believe the root cause is that we're getting the amount of space we need for the actual terminal pane itself, but then no accounting for the window frame. That window frame rounds up to roughly three characters tall on top/bottom, and one character on the left/right. If we accounted for those dimensions correctly when creating the window, both problems would go away. 

I believe the `IslandWindow` is doing this correctly, but `NonClientIslandWindow` is where the bug lies.

**Comment by fpqc** [comment]
@zadjii-msft I found the source of the issue.  It's padding, the scrollbar, and the tab UI.  I have tabs, scrollbar, and padding disabled, and I get pixel-perfect accuracy.

**Comment by daviox** [comment]
I'd say it's just tab UI, because I previously had a scrollbar set as hidden and padding set to 5px each. Disabling the tab UI solved the issue, at least in my case.

**Comment by zypA13510** [comment]
Not sure if it's worth mentioning, for me, it's always 2 rows smaller (instead of 3).

**Comment by zypA13510** [comment]
This has still not solved the scaling issue with padding (#4077).

I'm testing with terminal v0.10.781.0, which should include commit 0e33d86.

Under 150% DPI:
- padding 0 does not cause any issue, console size is at 80x24
- padding 50 results in a 76x22 console
- padding 100 and it becomes 72x20

Under 100% DPI:
- padding 0 is 80x24
- padding 100 is 80x24

**Comment by DHowett-MSFT** [comment]
Huh, thanks.
